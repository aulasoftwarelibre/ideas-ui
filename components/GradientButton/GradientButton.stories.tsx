import React from "react";

import { storiesOf } from "@storybook/react";
import { text } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";

import GradientButton from ".";

export default {
  title: "Gradient Button",
  component: GradientButton,
};

export const Default = () => (
  <GradientButton onClick={action("clicked")} >
      {text("Text", "I am a button")}
  </GradientButton>
)