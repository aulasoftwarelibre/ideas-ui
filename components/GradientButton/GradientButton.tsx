import React from "react";

import "./GradientButton.css";

export interface Props {
  onClick?: () => void;
}

export const GradientButton: React.FunctionComponent<Props> = ({
  onClick,
  children,
}) => {
  return (
    <button
      className="bg-gradient-l-gorange hover:bg-none hover:bg-gray text-white font-bold py-5 px-10 rounded-full"
      onClick={onClick}
    >
      {children}
    </button>
  );
};

export default GradientButton;
