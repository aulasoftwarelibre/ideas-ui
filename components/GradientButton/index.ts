import GradientButton from './GradientButton';

export default GradientButton;
