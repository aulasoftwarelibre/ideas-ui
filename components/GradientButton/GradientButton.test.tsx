import { text } from "@storybook/addon-knobs";
import { shallow } from "enzyme";
import React from "react";
import GradientButton from "./GradientButton";

describe("GradientButton", () => {
  const onClickMock = jest.fn();
  const component = shallow(
    <GradientButton onClick={onClickMock}>Gradient Button</GradientButton>
  );

  it("should be rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should be clickable", () => {
    component.simulate("click");

    expect(onClickMock).toHaveBeenCalled();
  });
});
