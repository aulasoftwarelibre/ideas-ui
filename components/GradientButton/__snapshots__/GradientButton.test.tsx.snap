// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradientButton should be rendered correctly 1`] = `ShallowWrapper {}`;
